serializer.rst

```diff
@@ -234,195 +454,1556 @@ You can also specify the context on a per-property basis::
+
+    Always make sure the ``DateTimeNormalizer`` is registered when
+    serializing the ``DateTime`` or ``DateTimeImmutable`` classes to avoid
+    excessive memory usage and exposing internal details.
+
+Built-in Normalizer
+...................
+
+Besides the normalizers registered by default (see previous section), the
+serializer component also provides some extra normalizers.You can register
+these by defining a service and tag it with :ref:`serializer.normalizer <reference-dic-tags-serializer-normalizer>`.
+For instance, to use the ``CustomNormalizer`` you have to define a service
+like:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            # if you're using autoconfigure, the tag will be automatically applied
+            Symfony\Component\Serializer\Normalizer\CustomNormalizer:
+                tags:
+                    # register the normalizer with a high priority (called earlier)
+                    - { name: 'serializer.normalizer', priority: 500 }
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <!-- if you're using autoconfigure, the tag will be automatically applied -->
+                <service id="Symfony\Component\Serializer\Normalizer\CustomNormalizer">
+                    <!-- register the normalizer with a high priority (called earlier) -->
+                    <tag name="serializer.normalizer"
+                        priority="500"
+                    />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+        use Symfony\Component\Serializer\Normalizer\CustomNormalizer;
+
+        return function(ContainerConfigurator $container) {
+            // ...
+
+            // if you're using autoconfigure, the tag will be automatically applied
+            $services->set(CustomNormalizer::class)
+                // register the normalizer with a high priority (called earlier)
+                ->tag('serializer.normalizer', [
+                    'priority' => 500,
+                ])
+            ;
+        };
+
+:class:`Symfony\\Component\\Serializer\\Normalizer\\CustomNormalizer`
+    This normalizer calls a method on the PHP object when normalizing. The
+    PHP object must implement :class:`Symfony\\Component\\Serializer\\Normalizer\\NormalizableInterface`
+    and/or :class:`Symfony\\Component\\Serializer\\Normalizer\\DenormalizableInterface`.
```

Comment:
I find it weird that this is the only built-in normalizer that is not registered by default. Is there a reason for this that we need to explain? (e.g. performance?)